localstack/services/kinesis/resource_providers/aws_kinesis_stream.py

```diff
@@ -0,0 +1,167 @@
+# LocalStack Resource Provider Scaffolding v2
+from __future__ import annotations
+
+from pathlib import Path
+from typing import Optional, Type, TypedDict
+
+import localstack.services.cloudformation.provider_utils as util
+from localstack.services.cloudformation.resource_provider import (
+    CloudFormationResourceProviderPlugin,
+    OperationStatus,
+    ProgressEvent,
+    ResourceProvider,
+    ResourceRequest,
+)
+
+
+class KinesisStreamProperties(TypedDict):
+    Arn: Optional[str]
+    Name: Optional[str]
+    RetentionPeriodHours: Optional[int]
+    ShardCount: Optional[int]
+    StreamEncryption: Optional[StreamEncryption]
+    StreamModeDetails: Optional[StreamModeDetails]
+    Tags: Optional[list[Tag]]
+
+
+class StreamModeDetails(TypedDict):
+    StreamMode: Optional[str]
+
+
+class StreamEncryption(TypedDict):
+    EncryptionType: Optional[str]
+    KeyId: Optional[str]
+
+
+class Tag(TypedDict):
+    Key: Optional[str]
+    Value: Optional[str]
+
+
+REPEATED_INVOCATION = "repeated_invocation"
+
+
+class KinesisStreamProvider(ResourceProvider[KinesisStreamProperties]):
+
+    TYPE = "AWS::Kinesis::Stream"  # Autogenerated. Don't change
+    SCHEMA = util.get_schema_path(Path(__file__))  # Autogenerated. Don't change
+
+    def create(
+        self,
+        request: ResourceRequest[KinesisStreamProperties],
+    ) -> ProgressEvent[KinesisStreamProperties]:
+        """
+        Create a new resource.
+
+        Primary identifier fields:
+          - /properties/Name
+
+
+
+        Create-only properties:
+          - /properties/Name
+
+        Read-only properties:
+          - /properties/Arn
+
+        IAM permissions required:
+          - kinesis:EnableEnhancedMonitoring
+          - kinesis:DescribeStreamSummary
+          - kinesis:CreateStream
+          - kinesis:IncreaseStreamRetentionPeriod
+          - kinesis:StartStreamEncryption
+          - kinesis:AddTagsToStream
+          - kinesis:ListTagsForStream
+
+        """
+        model = request.desired_state
+        kinesis = request.aws_client_factory.kinesis
+        if not model.get("Name"):
+            model["Name"] = util.generate_default_name(
+                stack_name=request.stack_name, logical_resource_id=request.logical_resource_id
+            )
+        if not model.get("ShardCount"):
+            model["ShardCount"] = 1
+
+        if not model.get("StreamModeDetails"):
+            model["StreamModeDetails"] = StreamModeDetails(StreamMode="ON_DEMAND")
+
+        kinesis.create_stream(
+            StreamName=model["Name"],
+            ShardCount=model["ShardCount"],
+            StreamModeDetails=model["StreamModeDetails"],
+        )
+
+        model["Arn"] = kinesis.describe_stream(StreamName=model["Name"])["StreamDescription"][
+            "StreamARN"
+        ]
+        return ProgressEvent(
+            status=OperationStatus.SUCCESS,
+            resource_model=model,
+            custom_context=request.custom_context,
+        )
```

Comment:
If I rembeber correctly the creation of a Kinesis Stream is an async process and Cloudformation won't return a success event until the Stream has an status active.

Reply:
Thanks for information. I'll make necessary changes.

Reply:
Done